mungegithub/mungers/mungerutil/util.go

```diff
@@ -16,14 +16,137 @@ limitations under the License.
 
 package mungerutil
 
-import "github.com/google/go-github/github"
+import (
+	"strings"
 
-// FilterValid filters a list of user data and returns only user data that contains valid username.
-func FilterValid(users []*github.User) (res []*github.User) {
-	for _, u := range users {
-		if u != nil && u.Login != nil {
-			res = append(res, u)
+	"github.com/google/go-github/github"
+)
+
+// ActiveUsers tracks Users involved in a github Issue
+type ActiveUsers struct {
+	assignees []string
+	author    string
+}
+
+func mergeUsers(userslists ...[]string) []string {
+	uniqueUsers := map[string]struct{}{}
+
+	// Fill-in map of all the users in each lists (remove duplicates)
+	for _, userslist := range userslists {
+		for _, user := range userslist {
+			uniqueUsers[user] = struct{}{}
+		}
+	}
+
+	// Get the list of unique users
+	users := []string{}
+	for user := range uniqueUsers {
+		users = append(users, user)
+	}
+	return users
+}
+
+// GetUser returns nil if the user is invalid, Login name otherwise
+func GetUser(user *github.User) *string {
+	if user == nil {
+		return nil
+	}
+	return user.Login
+}
+
+// GetUsersList returns the list of valid users
+func GetUsersList(users []*github.User) []string {
+	allUsers := []string{}
+
+	for _, user := range users {
+		validUser := GetUser(user)
+		if validUser != nil {
+			allUsers = append(allUsers, *validUser)
+		}
+	}
+	return mergeUsers(allUsers)
+}
+
+// CreateActiveUsersFromIssue creates a new ActiveUsers object from an issue's fields
+func CreateActiveUsersFromIssue(issue *github.Issue) *ActiveUsers {
+	return createActiveUsers(
+		issue.Assignee,
+		issue.Assignees,
+		issue.User,
+	)
+}
+
+// GetIssueAuthor returns the author of the issue
+func GetIssueAuthor(issue *github.Issue) string {
+	return CreateActiveUsersFromIssue(issue).Author()
+}
+
+// GetIssueAssignees returns the list of assignees
+func GetIssueAssignees(issue *github.Issue) []string {
+	return CreateActiveUsersFromIssue(issue).Assignees()
+}
+
+// GetIssueUsers returns all users of an issue
+func GetIssueUsers(issue *github.Issue) []string {
+	return CreateActiveUsersFromIssue(issue).AllUsers()
+}
+
+func createActiveUsers(assignee *github.User, assignees []*github.User, author *github.User) *ActiveUsers {
+	authorName := GetUser(author)
+	if authorName == nil {
+		authorName = new(string)
+	}
+
+	return &ActiveUsers{
+		assignees: mergeUsers(
+			GetUsersList(assignees),
+			GetUsersList([]*github.User{assignee})),
+		author: *authorName,
+	}
+}
+
+// MentionUsers adds a @ before each name in the list
+func MentionUsers(users []string) []string {
```

Comment:
consider making a userlist type with methods instead of having a bunch of functions that take a list of users.